--- gcc-9.4.0/gcc/c/c-typeck.c	2021-06-01 00:53:04.456471600 -0700
+++ gcc-9.4.0-patched/gcc/c/c-typeck.c	2022-06-21 13:10:34.492035100 -0700
@@ -2532,18 +2532,10 @@
 
       return ref;
     }
-  else if (should_suggest_deref_p (type))
-    {
-      /* Special-case the error message for "ptr.field" for the case
-	 where the user has confused "." vs "->".  */
-      rich_location richloc (line_table, loc);
-      /* "loc" should be the "." token.  */
-      richloc.add_fixit_replace ("->");
-      error_at (&richloc,
-		"%qE is a pointer; did you mean to use %<->%>?",
-		datum);
-      return error_mark_node;
-    }
+  else if (code == POINTER_TYPE)
+    return build_component_ref (loc, build_indirect_ref (loc, datum,
+                                                         RO_UNARY_STAR)
+                                , component, component_loc);
   else if (code != ERROR_MARK)
     error_at (loc,
 	      "request for member %qE in something not a structure or union",
